[chewbum] iP

chewbum opened the pull request:
# Chewbum's IP Pull Request  :rocket: :smiley:

> "Life is too complicated not to be orderly." – Martha Stewart

**Cleo Task Manager** helps you manage your tasks efficiently, freeing your mind to focus on the task at hand. It's
- *CLI-based*, perfect for keyboard warriors
- **Simple to learn**
- ~~Quick~~ **Blazingly fast** to use

## Getting Started:

1. Clone the repository from [GitHub](https://github.com/chewbum/ip).
2. Navigate to the repo directory and compile the code:
    ```bash
    javac -cp src\main\java\duke\Duke.java
    ```
3. Run Duke:
    ```bash
    java -cp src\main\java\duke\Duke
    ```

## Features:

- [x] Add and remove tasks
- [x] Mark tasks as done
- [x] List all tasks
- [x] Find tasks by keywords

## Usage Examples:

### Adding a task
You can add tasks of types todo, deadline, and event:

```java
todo read book
deadline return book /by 2020-10-15
event project meeting /at 2020-10-15 14:00

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
level 0

Commit:
Level 1 Echo

Commit:
Level-2 Add, List

Commit:
Level-3 Mark as Done

Commit:
Level-4 ToDos,Events,Deadlines

Commit:
Level-5

Commit:
Level-6

Commit:
Automated Testing

Commit:
Level- 7 Add TaskStorage

Commit:
Level 7 Integrated File Saving into Duke

Commit:
Merge branch 'branch-Level-7'

Commit:
Level-8 branch

Commit:
Merge branch 'branch-Level-8'

Commit:
Use More OOP

Commit:
Merge branch 'add-gradle-support'

Commit:
Add JUnit Tests

Commit:
Create a JAR File

Commit:
Add javadoc

Commit:
Add find feature

Commit:
Merge branch 'branch-Level-9'

Commit:
Added GUI

Commit:
A-Varags

timothysashimi commented on src/main/java/Command/ListCommand.java:
```diff
@@ -0,0 +1,10 @@
+package Command;
+import duke.TaskList;
+import duke.UI;
```
i think package names should start with small letters! eg. command

timothysashimi commented on src/main/java/Command/UnMarkCommand.java:
```diff
@@ -0,0 +1,25 @@
+public class UnMarkCommand extends Command {
+    private int taskIndex;
+
+    public UnMarkCommand(int taskIndex) {
+        this.taskIndex = taskIndex;
+    }
```
should it be unmarkCommand? If we follow camelcase convention

timothysashimi commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,84 @@
+     */
+    public void run() throws DukeException {
+        ui.showWelcomeMessage();
+        boolean isExit = false;
```
i like how you named ur boolean isExit!

timothysashimi commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,39 @@
+package duke;
+
+import java.time.LocalDate;
+;
```
why is there a random semicolon here?

timothysashimi reviewed (commented):
Looks good to me!

Some cosmestic changes requested

ashleyclx commented on src/main/java/Command/AddCommand.java:
```diff
@@ -0,0 +1,22 @@
+package Command;
```
package name shouldn't be capitalised

ashleyclx commented on src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,83 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class UI {
```
Since UI is an acronym, I think it would be better for it to be named Ui to follow PascalCase

ashleyclx commented on src/main/java/duke/TaskStorage.java:
```diff
@@ -0,0 +1,55 @@
+package duke;
+
+import java.io.*;
```
I think it would be better to list out the individual dependencies you wish to import instead of using a wildcard

ashleyclx reviewed (commented):
LGTM! Just a few nits to fix

Commit:
Add assertions to check index bounds in TaskList methods

Ensure that the index parameters for getTask, removeTask, and other index-based access methods in TaskList are within valid bounds. This prevents ArrayIndexOutOfBoundsException by catching invalid indices early in the development phase.

- Assert that index is >= 0 and < size of tasks list in getTask and removeTask.
- Similar checks added to any other method that uses an index to access the tasks list.

These assertions are intended for use in development and testing environments to catch and rectify errors early.

Commit:
Add CommandTypes enum to standardize command parsing

Introduce a CommandTypes enum to encapsulate the different types of commands the application can process. This change aims to improve code readability and maintainability by replacing string-based command checks with a more structured and type-safe approach. The enum includes a method to safely parse command strings, returning a default UNKNOWN type for unhandled cases, thereby enhancing error handling and future-proofing the command parsing mechanism.

chewbum committed:
Merge pull request #1 from chewbum/branch-A-Assertion

Add assertions to check index bounds in TaskList methods

Commit:
Merge branch 'master' into branch-A-CodeQuality

chewbum committed:
Merge pull request #2 from chewbum/branch-A-CodeQuality

Add CommandTypes enum to standardize command parsing

Commit:
C-Sort

chewbum committed:
Merge pull request #3 from chewbum/branch-A-Extension

C-Sort

Commit:
Improve code readability by renaming command package
Refactor command package name to lowercase

Commit:
Improved gui

Added title message to make the application more personalised

Commit:
Added screenshot of gui

Added screenshot into docs folder as per required

chewbum committed:
Update README.md

Commit:
Added JavaDoc comments

Added JavaDoc comments to more than half the classes and also did a bit
of refactoring to improve code quality

Commit:
Edited readME

Updated the user guide and tested out jar file

Commit:
Updated readme

Replaced template with my own user guide

chewbum committed:
Update README.md

chewbum committed:
Update README.md

chewbum committed:
Update README.md

chewbum committed:
Update README.md

chewbum committed:
Update README.md

Commit:
Update README.md

chewbum committed:
Merge pull request #6 from chewbum/branch-A-Assertions

Update README.md
